File: packages/vision-camera-pose-landmarks-plugin/ios/PoseLandmarksPlugin.swift
import VisionCamera

@objc(PoseLandmarksPlugin)
public class PoseLandmarksPlugin: FrameProcessorPlugin {
    private var mediaPipeFrameProcessor: MediaPipeFrameProcessor

    override public init(
        proxy: VisionCameraProxyHolder,
        options: [AnyHashable: Any]? = [:]
    ) {
        let numPoses = options?["numPoses"] as? Int
        let minPoseDetectionConfidence = options?["minPoseDetectionConfidence"] as? Float
        let minPosePresenceConfidence = options?["minPosePresenceConfidence"] as? Float
        let minTrackingConfidence = options?["minTrackingConfidence"] as? Float
        let modelPath = Bundle.main.path(
            forResource: "pose_landmarker_full",
            ofType: "task"
        )

        mediaPipeFrameProcessor = MediaPipeFrameProcessor(
            numPoses: numPoses,
            modelPath: modelPath,
            minTrackingConfidence: minTrackingConfidence,
            minPosePresenceConfidence: minPosePresenceConfidence,
            minPoseDetectionConfidence: minPoseDetectionConfidence
        )

        super.init(proxy: proxy, options: options)
    }

    override public func callback(
        _ frame: Frame,
        withArguments _: [AnyHashable: Any]?
    ) -> Any {
        mediaPipeFrameProcessor.process(frame)
    }
}

File: packages/vision-camera-pose-landmarks-plugin/android/src/main/java/com/visioncamera/poselandmarksplugin/PoseLandmarksPlugin.kt
package com.visioncamera.poselandmarksplugin

import com.mrousavy.camera.frameprocessors.Frame
import com.mrousavy.camera.frameprocessors.FrameProcessorPlugin
import com.mrousavy.camera.frameprocessors.VisionCameraProxy

class PoseLandmarksPlugin(proxy: VisionCameraProxy, options: Map<String, Any>?) :
    FrameProcessorPlugin() {
    private val mediaPipeProcessor: MediaPipeFrameProcessor

    init {
        val modelPath = "pose_landmarker_full.task"
        val numPoses = options?.get("numPoses") as? Double
        val minPoseDetectionConfidence = options?.get("minPoseDetectionConfidence") as? Double
        val minPosePresenceConfidence = options?.get("minPosePresenceConfidence") as? Double
        val minTrackingConfidence = options?.get("minTrackingConfidence") as? Double

        mediaPipeProcessor = MediaPipeFrameProcessor(
            proxy.context,
            modelPath,
            numPoses?.toInt(),
            minTrackingConfidence?.toFloat(),
            minPosePresenceConfidence?.toFloat(),
            minPoseDetectionConfidence?.toFloat()
        )
    }

    override fun callback(frame: Frame, params: Map<String, Any>?): Any {
        return mediaPipeProcessor.process(frame)
    }

    companion object {
        const val NAME = "detectPoseLandmarks"
    }
}
